homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+
+        if cached_state is None:
+            final_state = entity.state == 'on'
+            final_brightness = entity.attributes.get(
+                ATTR_BRIGHTNESS, 255 if final_state else 0)
+        else:
+            final_state, final_brightness = cached_state
+
+        json_response = entity_to_json(entity, final_state, final_brightness)
+
+        return Response(json.dumps(json_response), mimetype='application/json')
+
+    # pylint: disable=no-self-use
+    def hue_api_create_user(self):
+        """Handle requests to create a new user for the local bridge."""
+        from flask import Response, request, abort
+
+        request_json = request.get_json(force=True)
+
+        if 'devicetype' not in request_json:
+            abort(500)
+
+        json_response = [{'success': {'username': '12345678901234567890'}}]
+
+        return Response(json.dumps(json_response), mimetype='application/json')
+
+
+def parse_hue_api_put_light_body(request_json, entity):
+    """Parse the body of a request to change the state of a light."""
+    if 'on' in request_json:
+        if not isinstance(request_json['on'], bool):
+            return None
+
+        if request_json['on']:
+            # Echo requested device be turned on
+            brightness = 100
+            report_brightness = False
+            result = True
+        else:
+            # Echo requested device be turned off
+            brightness = 0
+            report_brightness = False
+            result = False
+
+    if 'bri' in request_json:
+        # Make sure the entity actually supports brightness
+        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+        if (entity_features & SUPPORT_BRIGHTNESS) == SUPPORT_BRIGHTNESS:
+            try:
+                # Clamp brightness from 0 to 255
+                brightness = max(0, min(int(request_json['bri']), 255))
+            except ValueError:
+                return None
+
+            report_brightness = True
+            result = True if brightness > 0 else False
+
+    return (result, brightness) if report_brightness else (result, None)
+
+
+def entity_to_json(entity, is_on=None, brightness=None):
+    """Convert an entity to its JSON representation."""
+    if is_on is None:
+        is_on = entity.state == "on"
+
+    if brightness is None:
+        brightness = 255 if is_on else 0
+
+    name = entity.attributes.get(
+        ATTR_ECHO_FRIENDLY_NAME, entity.attributes['friendly_name'])
+
+    return {
+        'state':
+        {
+            'on': is_on,
+            'bri': brightness,
+            'hue': 0,
+            'sat': 0,
+            'effect': 'none',
+            'ct': 0,
+            'alert': 'none',
+            'reachable': True
+        },
+        'type': 'Dimmable light',
+        'name': name,
+        'modelid': 'LWB004',
+        'manufacturername': 'Philips',
+        'uniqueid': entity.entity_id,
+        'swversion': '66012040'}

```

Comment:
Some of these fields aren't required to work with Alexa. Let's remove any fields that it doesn't absolutely require so that we can fake as little as possible.